### ISSUE TITLE
Remove dependencies on k8s.io/kubernetes

### ISSUE THREAD

**Issue by damemi** [kubernetes-sigs/descheduler#200]
There are several places in the code where we are importing `k8s.io/kubernetes`

```
$ grep -rni "k8s.io/kubernetes" *
pkg/descheduler/pod/pods_test.go:24:	"k8s.io/kubernetes/pkg/apis/scheduling"
pkg/descheduler/pod/pods.go:24:	api "k8s.io/kubernetes/pkg/apis/core"
pkg/descheduler/pod/pods.go:25:	"k8s.io/kubernetes/pkg/apis/core/v1/helper/qos"
pkg/descheduler/pod/pods.go:26:	"k8s.io/kubernetes/pkg/kubelet/types"
pkg/descheduler/strategies/lownodeutilization.go:26:	helper "k8s.io/kubernetes/pkg/api/v1/resource"
pkg/descheduler/strategies/pod_antiaffinity.go:29:	priorityutil "k8s.io/kubernetes/pkg/scheduler/algorithm/priorities/util"
pkg/utils/predicates.go:25:	v1helper "k8s.io/kubernetes/pkg/apis/core/v1/helper"
pkg/utils/predicates.go:30:// k8s.io/kubernetes/plugin/pkg/scheduler/algorithm/predicates/
test/e2e/e2e_test.go:29:	"k8s.io/kubernetes/pkg/api/testapi"
```

At a glance many of these seem to be importing internal APIs, for possibly dubious reasons (such as [here](https://github.com/kubernetes-sigs/descheduler/blob/dd54f1a6560f2945502af3b5347beaeceb1449a2/pkg/descheduler/pod/pods.go#L24), where this is only used to access 2 constants, both which are available in `k8s.io/api`). Some of these changes may require minor refactors to `k8s.io/kubernetes` (such as moving constants that never got copied to the external APIs)

I think we should work to break these dependencies on `k8s.io/kubernetes`, as I don't think it is intended to be imported by external tools and this will make the switch to go.mod simpler (as well as reduce our transitive dependencies)

**Comment by aveshagarwal** [comment]
@damemi we dont have to use k8s.io/kubernetes` and also we should stop using internap apis and replaced them with external exported apis.

**Comment by seanmalloy** [comment]
@damemi I think this issue can be closed now that #216 has been merged. Is there anything else that need to be done before we close this one out?

**Comment by damemi** [comment]
@seanmalloy nope, thought it would get closed automatically. Thanks for reminding me
/close
